doc/spec/json.md

```diff
@@ -0,0 +1,92 @@
+# Docker Distribution JSON Canonicalization
+
+## Introduction
+
+To provide consistent content hashing of JSON object throughout Docker
+Distribution APIs, a canonical JSON format has been defined. The
+canonicalization is balance between conistency and readability. The format
+also provides consistent output for effective caching.
+
+## Rules
+
+Compliant JSON should conform to the following rules:
+
+1. All generated JSON should comply with [RFC
+   7159](http://www.ietf.org/rfc/rfc7159.txt).
+2. Resulting "JSON text" shall always be encoded in UTF-8.
+3. Unless a canonical key order is defined for a particular schema, object
+   keys shall always appear in lexically sorted order.
+4. The colon (":") shall always be followed by a single space.
+5. An indent of three spaces ("   ") shall be used.
```

Comment:
I was going to ask the same thing. It seems like it would lead to less room for error if you don't allow pretty printing and instead have the json as compact as possible.